### ISSUE TITLE
Add support for Tag Expressions

### ISSUE THREAD

**Issue by aslakhellesoy** [cucumber/cucumber-js#623]
## Summary

[Tag Expressions](http://docs.cucumber.io/tag-expressions/) is a more intuitive way to specify tags in tagged hooks and on the command line's `--tags` argument.
## Expected Behavior

Tagged Hooks should accept a single string:

``` JavaScript
Before("@foo and not @bar", function() {
})
```

Likewise, the `--tags` command line argument should allow the same. (Specifying it more than once should cause an error)

```
--tags "@foo and not @bar"
```
## Context & Motivation

The current mechanism to specify `or` with comma and `and` with multiple `--tags` is incredibly awkward. Tag Expressions are intuitive!

**Comment by adamlacoste** [comment]
I came here today to request just such a feature.  I have a use case where I want to trigger multiple separate cucumber executions for various browsers.  My features specs might be tagged with one, many or none of these browsers; in cases where no browser tag is present, I want that feature/scenario to be treated as a valid candidate for any browser test.  So what I needed was a cucumber command line argument that would let me specify, for instance, @browser1 OR ~@browser2 AND ~@browser3  (it either has the specified browser tag, or has no other browser tags).  Unfortunately it's not possible with the current system.  Eagerly awaiting this feature.

**Comment by charlierudolph** [comment]
Since this is going to be a breaking change, I'm going to put this in 2.0 which I'm working on at the moment and is full rewrite of the cucumber-js internals.

**Comment by charlierudolph** [comment]
Available in `2.0.0-rc.1`
